### PR TITLE
sigstore, test: add CircleCI credential detection

### PR DESCRIPTION
**Pull request by woodruffw**
See #31.

Signed-off-by: William Woodruff <william@trailofbits.com>

### CODE REVIEW

**Comment by woodruffw** [comment]
Yep, correct. It looks like CircleCI uses a similar permission style as GitHub, but skips the indirection of a request.

File: sigstore/_internal/oidc/ambient.py
Comment (woodruffw) [reply sigstore/_internal/oidc/ambient.py:114]: Not sure how this got mis-threaded, but I responded here: https://github.com/sigstore/sigstore-python/pull/72#issuecomment-1119810936

(So yes: it's the actual identity token 🙂 )

Comment (woodruffw) [reply sigstore/_internal/oidc/ambient.py:114]: From their docs:

> In CircleCI jobs that use at least one context, the OpenID Connect ID token is available in the environment variable CIRCLE_OIDC_TOKEN.

**Review by di — DISMISSED** [review]
This LGTM but I want to add a simple integration test for CircleCI before merging. I don't have the ability to do this for this repo anymore though, so hang tight while I figure out who can.

**Comment by woodruffw** [comment]
Sounds good!

File: .circleci/config.yml
Comment (woodruffw) [inline .circleci/config.yml:6]: N.B.: We might need a context declared here to get the ambient OIDC token: https://circleci.com/docs/2.0/openid-connect-tokens/#openid-connect-id-token-availability

**Comment by di** [comment]
This seems like a dealbreaker. From https://circleci.com/docs/2.0/openid-connect-tokens/#format-of-the-openid-connect-id-token

> `aud`: The audience. Currently, this is a fixed value "<organization-id>", a string containing a UUID that identifies the job’s project’s organization.

**Comment by di** [comment]
Marking this as blocked on https://circleci.canny.io/cloud-feature-requests/p/customizable-audience-claim-in-oidc-tokens, please upvote that feature request if you need this feature.

**Comment by woodruffw** [comment]
Oh well; that's too bad. At least the changeset here is pretty small, so we'll be able to move this along rapidly once that gets unblocked.

**Comment by di** [comment]
Given the slow movement here, dropping this out of the 1.0 milestone.

**Comment by jerdog** [comment]
CircleCI has [just added](https://circleci.canny.io/cloud-feature-requests/p/customizable-audience-claim-in-oidc-tokens) customizable audience claims in OIDC tokens =) 

https://circleci.com/docs/api/v2/index.html#tag/OIDC-Token-Management

**Comment by woodruffw** [comment]
@jerdog fantastic, thanks for letting us know!

This PR is pretty stale at this point, but I'll see about refreshing it.

**Comment by woodruffw** [comment]
Upstream tracking: https://github.com/di/id/issues/61

**Pull request closed without merging**
